### DISCUSSION TITLE
Tasmota Scripting: How to capture Button Multipress Values?

### DISCUSSION THREAD

**Discussion by jkwim** [arendst/Tasmota#18536] [Q&A] [answered]
I need some help to understand how to capture Button Multipress Values in a script.

I want to capture the following multipresses:

I have turned ON `SetOption73`.

```
11:00:12.614 MQT: stat/xxxx/RESULT = {"Button1":{"Action":"SINGLE"}}                ENUMERATION INDEX=10
11:00:21.218 MQT: stat/xxxx/RESULT = {"Button1":{"Action":"DOUBLE"}}               ENUMERATION INDEX=11
11:00:27.231 MQT: stat/xxxx/RESULT = {"Button1":{"Action":"TRIPLE"}}                  ENUMERATION INDEX=12
11:01:35.943 MQT: stat/xxxx/RESULT = {"Button1":{"Action":"QUAD"}}                  ENUMERATION INDEX=13
11:02:22.044 MQT: stat/xxxx/RESULT = {"Button1":{"Action":"HOLD"}}                   ENUMERATION INDEX=3
11:02:23.151 MQT: stat/xxxx/RESULT = {"Button1":{"Action":"CLEAR"}}
```
the function `sw[0]` does not seem to return the above index values but only the value 1.

Here is the script:

```
>D
buttonstate=0
 
>E
print "-"
buttonstate=sw[0]
print buttonstate %buttonstate%
```
Output:
```
15:48:34.469 "-"
15:48:34.472 buttonstate 1.00
15:48:34.478 MQT: stat/xxxx/RESULT = {"Button1":{"Action":"DOUBLE"}}
15:48:34.481 "-"
15:48:34.483 buttonstate 1.00
```

**Comment by jkwim** [comment]
@gemu2015

Any suggestions please?

**Reply by gemu2015** [reply]
there is no direct support for button multi presses.
you must catch and decode the corresponding JSON event.

i just added an entry for FUNC_BUTTON_MULTI_PRESSED in section >b 
with 
#define  USE_BUTTON_EVENT

**Reply by gemu2015** [reply, marked as answer]
i never used multi presses but you may try this:
```
>D
action=""
>E
action=Button1#Action
switch action
case "SINGLE"
print single press
case "DOUBLE"
print double press
;etc
ends
action=""
```
instead of >E may also try >b (with my latest fork)

**Reply by jkwim** [reply]
> i never used multi presses but you may try this:
> 
> ```
> >D
> action=""
> >E
> action=Button1#Action
> switch action
> case "SINGLE"
> print single press
> case "DOUBLE"
> print double press
> ;etc
> ends
> action=""
> ```
> 
> instead of >E may also try >b (with my latest fork)

Works great!

Thanks a lot!
